src/NumericalAlgorithms/SphericalHarmonics/StrahlkorperFunctions.hpp

```diff
@@ -348,4 +348,25 @@ template <typename Fr>
 void normal_one_form(const gsl::not_null<tnsr::i<DataVector, 3, Fr>*> one_form,
                      const tnsr::i<DataVector, 3, Fr>& dx_radius,
                      const tnsr::i<DataVector, 3, Fr>& r_hat);
+
+/// @{
+/*!
+ * The linear least squares fit of the polynomial of order 3
+ * given a std::vector of Strahlkorpers to the ylm coefficients.
+ * Assumes the the l_max and m_max of each Strahlkorper are the same,
+ * and the returned vector consists of 2 * (l_max) * (m_max) (the number of
+ * ylm coefficients) `std::array<double, 4>`s, each of which consists of the
+ * four coefficients that define the best fit cubic to each ylm coefficient
+ * of the Strahlkorper as a function of time.
+ *
+ * \param times The time corresponding to each strahlkorper to be fit to.
+ * \param strahlkorpers The Strahlkorper surfaces which consists of a set
+ * of ylm coefficients corresponding to the shape of the strahlkorper at a
+ * particular time.
+ */
+template <typename Fr>
+std::vector<std::array<double, 4>> fit_ylm_coeffs(
+    const DataVector& times,
```

Comment:
why are the times in a DataVector?

Reply:
Hi @kidder this is because I use the linear least squares function which takes DataVector as input. Is that ok or should I set `times` to be something else?

Reply:
that is okay